eras/alonzo/impl/src/Cardano/Ledger/Alonzo/TxWits.hs

```diff
@@ -494,7 +494,15 @@ instance
       dec = do
         entries <- fmap sequence
           . decodeList
-          . decodeRecordNamed "redeemer" (const 4)
+          . decodeRecordNamed
+            "redeemer"
+            ( const
+                ( 2
+                    +
+                    {- This is a dummy value, but it enforces getting the correct listLen for the FromCBORGroup of RdmrPtr -}
+                    fromIntegral (listLen (RdmrPtr Spend 1))
+                )
+            )
```

Comment:
It is wrong to construct a dummy value, because you do have access to the actual value:
```suggestion
            (\ (rdmPtr, _) -> fromIntegral (listLen rdmPtr) + 2)
```

Reply:
I tried this earlier, but the type there happens to be `Int -> Annotator (RdmrPtr, (Data era, ExUnits)) -> Int`. So, even `(\(fmap fst . runAnnotator -> rdmrPtr) -> fromIntegral (listLen rdmrPtr) + 2)` does not work, as the `FullByteString -> a` remains a `Functor` that expects its argument to be supplied during decoding, I presume?

Reply:
That is a fairly easy problem to solve. All you gotta do is push the `decodeRecordName` inside where `rdmrPtr` hasn't been pushed inside an `Annotator`. Here is the full working solution for you:


```haskell

instance Era era => FromCBOR (Annotator (RedeemersRaw era)) where
  fromCBOR = do
    entries <- fmap sequence . decodeList $ decodeAnnElement
    pure $ RedeemersRaw . Map.fromList <$> entries
    where
      decodeAnnElement :: forall s. Decoder s (Annotator (RdmrPtr, (Data era, ExUnits)))
      decodeAnnElement = do
        (rdmrPtr, dat, ex) <- decodeElement
        let f x y z = (x, (y, z))
        pure $ f rdmrPtr <$> dat <*> pure ex
      decodeElement :: forall s. Decoder s (RdmrPtr, Annotator (Data era), ExUnits)
      decodeElement = do
        decodeRecordNamed
          "Redeemer"
          (\(rdmrPtr, _, _) -> fromIntegral (listLen rdmrPtr) + 2)
          $ (,,) <$> fromCBORGroup <*> fromCBOR <*> fromCBOR
```
Just copy an paste it and it will work